### DISCUSSION TITLE
Persistent values across triggered flows

### DISCUSSION THREAD

**Discussion by DigNZ** [directus/directus#15249] [Archive]
Not sure how possible this is but here is my flow that I want to be able to do:

1. Cron based trigger
2. Authenticate with a web service that returns a session token
3. Pull an array of data from the local database
4. For each element in that array make an API call using the session token from step 2

In order to make it work in a loop I pass the array to a Trigger Flow node which then calls that 'Another Flow' for each item in the array. But they don't have the session token. I don't really want to authenticate multiple times (putting that step inside the Another Flow) is there a way that I can pass that data? Also is there a way to run them in series instead of parallel I don't want to rate limit the API.

Any advice gratefully recieved!

**Comment by rijkvanzanten** [comment]
Ideally, there'd be a way to pass more "contextual" data to those flow executions when it runs the flow for each of the items in the array 🤔 I guess you could already do that by passing a more tailored array of data where you effectively put the token to use in the array of objects, like:

```
[
  { token: 'abc', item: {...} },
  { token: 'abc', item: {...} },
  { token: 'abc', item: {...} }
]
```

obviously not the neatest as we know that the token will be the same for every item, but it does mean that the individual flow executions have access to the token, as they will be called with `{ token: 'abc', item: {...} }` as the trigger payload.

Alternatively, you could save the token in a temporary row in the database, and read it from the database in the execution of the nested flow, and then clean out that row afterwards in the parent flow. 

As for configuring sequential vs parallel, and adding some sort of rate limiting / exponential backoff, that's currently not possible in the "execute flow" operation yet. Make sure to open a new feature request for that, so we can gauge community interest and start working on an actual implementation 😄

**Reply by LasseRosenow** [reply]
> I guess you could already do that by passing a more tailored array of data where you effectively put the token to use in the array of objects

I have a similar problem. Do i understand it correctly, that is is probably only possible with a script block?

Because somehow these tokens need to be put into the array

**Reply by rijkvanzanten** [reply]
Script block is probably the easiest way to set that up yeah!
